### PR TITLE
refactor: restrict data source reachability Lambda role

### PR DESCRIPTION
**Pull request by Bingjiling**
Issue #, if available:

Description of changes:
* Remove unused role envStatusPoll
* Restrict role dataSourceReachability
* Add permission boundary for dataSourceReachability
* Verified change with data sources integ tests
* Manually tested 2 data source reachability Lambdas

Checklist:

- [x] Have you successfully deployed to an AWS account with your changes?
- [ ] Have you written new tests for your core changes, as applicable?
- [x] Have you successfully tested with your changes locally?
- [ ] Have you updated openapi.yaml if you made updates to API definition (including add, delete or update parameter and request data schema)?
- [ ] If you had to run manual tests, have you considered automating those tests by adding them to [end-to-end tests](../main/end-to-end-tests/README.md)?

AS review ticket id:

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: main/solution/backend/config/infra/cloudformation.yml [hunk 1]
```diff
@@ -1949,8 +1917,6 @@ Outputs:
     Value: !GetAtt RoleWorkflowLoopRunner.Arn
   OpenDataScrapeHandlerRoleArn:
     Value: !GetAtt RoleOpenDataScrapeHandler.Arn
-  EnvStatusPollHandlerRoleArn:
-    Value: !GetAtt RoleEnvStatusPollHandler.Arn
```
Comment (jn1119) [inline main/solution/backend/config/infra/cloudformation.yml:1953]: Do we need to update this configuration as well: https://github.com/awslabs/service-workbench-on-aws/blob/4b5928ebbbd837d2e20781a213c2fcb630fbdd95/main/solution/backend/serverless.yml#L73-L75

**Review by jn1119 — COMMENTED** [review]
LGTM overall, just had one question

File: main/solution/backend/config/infra/cloudformation.yml (hunk 1 above)
Comment (Bingjiling) [reply main/solution/backend/config/infra/cloudformation.yml:1953]: Good catch!

File: main/solution/backend/config/infra/cloudformation.yml [hunk 2]
```diff
@@ -743,49 +743,35 @@ Resources:
...
-
-    # IAM Role for the envStatusPollHandler Function
-  RoleEnvStatusPollHandler:
-    Type: 'AWS::IAM::Role'
+  PolicyDataSourceReachabilityHandler:
+    Type: AWS::IAM::ManagedPolicy
     Properties:
-      AssumeRolePolicyDocument:
-        Version: '2012-10-17'
+      Description: Grant ClouWatch access for logging and ddb access to studies and accounts table
```
Comment (SanketD92) [inline main/solution/backend/config/infra/cloudformation.yml:749]: `DsAccounts` instead of `accounts`

**Review by jn1119 — APPROVED** [review]
LGTM

File: main/solution/backend/config/infra/cloudformation.yml (hunk 2 above)
Comment (Bingjiling) [reply main/solution/backend/config/infra/cloudformation.yml:749]: Gotcha! I'll address it as part of the next PR.

**Pull request merged**